src/go/rpk/pkg/cli/registry/schema/list.go

```diff
@@ -0,0 +1,125 @@
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/schemaregistry"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/sr"
+	"github.com/twmb/types"
+)
+
+type schemaResponse struct {
+	Subject string `json:"subject" yaml:"subject"`
+	Version int    `json:"version,omitempty" yaml:"version,omitempty"`
+	ID      int    `json:"ID,omitempty" yaml:"ID,omitempty"`
+	Type    string `json:"type,omitempty" yaml:"type,omitempty"`
+	Err     string `json:"error,omitempty" yaml:"error,omitempty"`
+}
+
+func newListCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var deleted bool
+	cmd := &cobra.Command{
+		Use:     "list [SUBJECT...]",
+		Aliases: []string{"ls"},
+		Short:   "List the schemas for the requested subjects, or list all schemas",
+		Run: func(cmd *cobra.Command, subjects []string) {
+			f := p.Formatter
+			if h, ok := f.Help([]schemaResponse{}); ok {
+				out.Exit(h)
+			}
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			cl, err := schemaregistry.NewSRClient(fs, p)
+			out.MaybeDie(err, "unable to initialize schema registry client: %v", err)
+
+			if len(subjects) == 0 {
+				subjects, err = cl.Subjects(cmd.Context(), sr.HideShowDeleted(deleted))
+				out.MaybeDieErr(err)
+			}
+
+			type res struct {
+				subject string
+				ss      []sr.SubjectSchema
+				err     error
+			}
+			var (
+				wg      sync.WaitGroup
+				mu      sync.Mutex
+				results []res
+			)
+			for i := range subjects {
+				subject := subjects[i]
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					ss, err := cl.Schemas(cmd.Context(), subject, sr.HideShowDeleted(deleted))
+					mu.Lock()
+					defer mu.Unlock()
+					results = append(results, res{
+						subject: subject,
+						ss:      ss,
+						err:     err,
+					})
+				}()
+			}
+			wg.Wait()
+
+			types.Sort(results)
+
+			// We use literal here to display an empty array when unmarshalling
+			// an empty list.
+			response := []schemaResponse{}
+			isText := f.IsText()
+			var tw *out.TabWriter
+			if isText {
+				tw = out.NewTable("subject", "version", "id", "type", "error")
+				defer tw.Flush()
+			}
+			for _, res := range results {
+				if res.err != nil {
+					sc := schemaResponse{Subject: res.subject, Err: res.err.Error()}
+					if isText {
+						tw.PrintStructFields(sc)
+					} else {
+						response = append(response, sc)
+					}
+					continue
+				}
+				for _, s := range res.ss {
+					sc := schemaResponse{
+						Subject: s.Subject,
+						Version: s.Version,
+						ID:      s.ID,
+						Type:    s.Type.String(),
+					}
+					if isText {
+						tw.PrintStructFields(sc)
+					} else {
+						response = append(response, sc)
+					}
+				}
+			}
+			if _, _, s, err := f.Format(response); !isText {
+				out.MaybeDie(err, "unable to print in the required format %q: %v", f.Kind, err)
+				out.Exit(s)
+			}
```

Comment:
Same comment I have in schema.go -- can the struct building logic be above the format / print logic? It keeps the logic a bit easier to follow.